Backport to glib-2.76.4 of patch submitted & accepted upstream for 2.83.0 in:
https://gitlab.gnome.org/GNOME/glib/-/merge_requests/4351

From 5ff042914700b50f8e5d58648a8cec9a76fa3540 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Subject: [PATCH 1/5] glib-unix: Fix build of safe_fdwalk() on Solaris

The refactoring done by commit 168fd4f2b3dbaa100ace1264462ccc6fc33ea39c
lost the definition of the open_max variable used in the Solaris ifdefs.

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 glib/glib-unix.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/glib/glib-unix.c b/glib/glib-unix.c
index 932c230d6..6b34234b8 100644
--- a/glib/gspawn.c
+++ b/glib/gspawn.c
@@ -1481,6 +1481,7 @@
  * fcntl(fd, F_PREVFD)
  * - return highest allocated file descriptor < fd.
  */
+  gint open_max;
   gint fd;
   gint res = 0;
 
-- 
2.45.2

